pkg/operator/controller/ingress/controller.go

```diff
@@ -95,6 +94,10 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err := c.Watch(&source.Kind{Type: &corev1.Service{}}, enqueueRequestForOwningIngressController(config.Namespace)); err != nil {
 		return nil, err
 	}
+	// Add watch for pods specifically for ensuring we know when pods get deleted for ensuring ingress deletion.
+	if err := c.Watch(&source.Kind{Type: &corev1.Pod{}}, enqueueRequestForOwningIngressController(config.Namespace)); err != nil {
```

Comment:
Is it okay that we reconcile on pod updates? It will generally increase the number of reconcilations, but is needed to reconcile again when the pods disappear.

The other option, eluded to below, is to reconcile on `ensureIngressDeleted` failure (or errors exist). If we did this, then arguably there is no need for watching the pods.